src/sagemaker_training/environment.py

```diff
@@ -358,6 +358,26 @@ def num_cpus():  # type: () -> int
     return multiprocessing.cpu_count()
 
 
+def validate_smddp_smddpmprun():  # type: () -> bool
+    """Whether smddp and smddpmprun are installed.
+
+    Returns:
+        bool: True if both are installed
+    """
+    try:
+        import smdistributed.dataparallel.torch.torch_smddp  # noqa: F401 # pylint: disable=unused-import
```

Comment:
This makes the check more complete and strict. I am thinking removing it should not break the logic.

Reply:
Will import initialize something here, can that cause issues?

Reply:
Yes, I think that it is a bit risky to import SMDDP at this layer. Today nothing happens, but in the future we may decide to initialize some services when the library is imported.

At this layer, we only care about SMDDP being installed, which is implicitly inferred since smddpmprun is installed as part of SMDDP installation process.

Reply:
Yes, agreed. Removed it. And also since the existences of `torch_smddp` and `smddpmprun` are consistent, checking one would be sufficient.